src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/service/LocationService.kt

```diff
@@ -1068,6 +1069,34 @@ class LocationService(
     }.sortedBy { it.pathHierarchy }
   }
 
+  fun getCellAttributes(id: UUID): List<CellAttributes> {
+    val cell = cellLocationRepository.findById(id).getOrNull()
+      ?: throw LocationNotFoundException(id.toString())
+
+    val specialistCellTypes = cell.specialistCellTypes
+      .map {
+        CellAttributes(
+          it.specialistCellType,
+          it.specialistCellType.description,
+        )
+      }
+
+    val legacyAttributes = cell.attributes.filter { it.attributeType == ResidentialAttributeType.LOCATION_ATTRIBUTE }
+      .map {
+        CellAttributes(
+          it.attributeValue,
+          it.attributeValue.description,
+        )
+      }
+
+    if (specialistCellTypes.size > 0) {
```

Comment:
Sorry @GurnankCheema think you mis understood this part.  You need to check if this prison where the location resides in turned "on" - check the `ActivePrisonConfig` if the prison matches the prison_id - *** is all